File: javascript/696.js
/**
 * @param {string} s
 * @return {number}
 */
var countBinarySubstrings = function(s) {
    let atual = 1;
    let anterior = 0;
    let total = 0;

    for (let i = 0; i < s.length - 1; i++) {
        if (s[i] == s[i + 1]) {
            ++atual;
        } else {
            total += Math.min(atual, anterior);
            anterior = atual;
            atual = 1;
        }
    }

    total += Math.min(atual, anterior);
    
    return total;
};

File: kotlin/696.kt
class Solution {
    fun countBinarySubstrings(s: String): Int {
        var atual = 1
        var anterior = 0
        var resposta = 0

        for (i in 0 until s.length - 1) {

            if (s[i] == s[i + 1]){
                ++atual
            } else {
               resposta += minOf(atual, anterior)
               anterior = atual
               atual = 1
            }
        }

        resposta += minOf(atual, anterior)

        return resposta
    }
}
